consensus/polybft/consensus_runtime.go

```diff
@@ -1007,6 +1009,12 @@ func (c *consensusRuntime) BuildCommitMessage(proposalHash []byte, view *proto.V
 
 // StartRound starts a new round with the specified view
 func (c *consensusRuntime) StartRound(view *proto.View) error {
+	if view.Round > 0 {
+		c.config.txPool.ReinsertProposed()
+	} else {
+		c.config.txPool.ClearProposed()
+	}
```

Comment:
It might be more safe to keep it this way. There is a check within those methods whether the proposed queue contains elements. If not then method returns immediately.